src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx

```diff
@@ -256,6 +258,7 @@ export function SavedQueryManagementComponent({
                   onClose={() => openSavedQueryFlyout?.close().then()}
                   onQueryOpen={onLoad}
                   handleQueryDelete={handleDelete}
+                  queryStringManager={queryStringManager}
```

Comment:
Nit: Instead of prop drilling `queryStringManager` throughout, `savedQueryService` should ideally have been responsible for templates too.